assets/src/web-stories-block/block/api/apiProvider.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useMemo } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../../dashboard/app/config';
+import { createContext } from '../../../design-system';
+import dataAdapter from '../../../dashboard/app/api/wpAdapter';
+import useStoryApi from './useStoryApi';
+import useUsersApi from './useUserApi';
+
+export const StoriesBlockApiContext = createContext({ state: {}, actions: {} });
+
+export default function StoriesBlockApiProvider({ children }) {
+  const { api, editStoryURL } = useConfig();
+
+  const { api: usersApi, authorSuggestions } = useUsersApi(dataAdapter, {
+    usersApi: api.users,
+  });
+
+  const { stories, api: storyApi } = useStoryApi(dataAdapter, {
```

Comment:
We don't need this whole `dataAdapter` logic in our Gutenberg blocks. We can use `@wordpress/api-fetch` directly in `useStoryApi`.

One file less to load from dashboard.

Reply:
I've just kept the required method in the `dataAdapter`, kept it this way so that we could easily update/modify it if we try to dedupe it later.